tests/rptest/scale_tests/tiered_storage_stability_test.py

```diff
@@ -294,3 +287,35 @@ def stage_block_s3(self):
         self.logger.info(f"Waiting for S3 errors to cease")
         wait_until(lambda: self._cloud_storage_no_new_errors(self.redpanda, self.logger),
                        timeout_sec=600, backoff_sec=20)
+
+    def stage_decommission_and_add(self):
+        node, node_id, node_str = self.get_node(1)
+        def partitions_in_nt(self):
+            try:
+                parts = self.redpanda.partitions(self.topic_name)
+            except StopIteration:
+                return 0
+            n = sum([1 if r.account==node.account else 0 for p in parts for r in p.replicas])
+            self.logger.debug ( f"Partitions in the node-topic: {n}" )
+            return n
+        nt_partitions_before = partitions_in_nt(self)
+
+        self.logger.info(f"Decommissioning node {node_str}, partitions: {nt_partitions_before}")
+        decomm_time = time.monotonic()
+        admin = self.redpanda._admin
+        admin.decommission_broker(node_id)
+        waiter = NodeDecommissionWaiter(self.redpanda, node_id, self.logger, progress_timeout=120)
+        waiter.wait_for_removal()
+        self.redpanda.stop_node(node)
+        assert partitions_in_nt(self)==0
+        decomm_time = time.monotonic() - decomm_time
+
+        self.logger.info(f"Adding a node")
+        self.redpanda.clean_node(node, preserve_logs=True, preserve_current_install=True)
+        self.redpanda.start_node(node, auto_assign_node_id=False, omit_seeds_on_idx_one=False)
+        wait_until(self.redpanda.healthy, timeout_sec=600, backoff_sec=1)
+        new_node_id = self.redpanda.node_id(node, force_refresh=True)
+
+        self.logger.info(f"Node added, new node_id: {new_node_id}, waiting for {int(nt_partitions_before/2)} partitions to move there in {int(decomm_time)} s")
+        wait_until(lambda: partitions_in_nt(self)>nt_partitions_before/2, timeout_sec=max(60,decomm_time), backoff_sec=2)
+        self.logger.info(f"{partitions_in_nt(self)} partitions moved")
```

Comment:
nit: spacing seems wonky here and above

Reply:
not sure what spacing do you mean... formatted